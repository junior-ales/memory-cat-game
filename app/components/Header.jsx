import React from 'react';
import { compose } from 'ramda';
import { addTodo } from '../actions';

const Header = ({ title, onSave }) => (
  <header>
    <h1>A Todo List: { title }</h1>
    <button onClick={() => onSave('foobar')}>Add</button>
  </header>
);

Header.propTypes = { // eslint-disable-line
  title: React.PropTypes.string.isRequired,
  onSave: React.PropTypes.func.isRequired
};

const extractProps = props => ({
  title: props.title,
  onSave: props.dispatch(addTodo)
});

export default compose(Header, extractProps);
