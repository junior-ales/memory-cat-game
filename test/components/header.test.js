/* eslint-disable */
import renderer from 'react-test-renderer';
import React from 'react';
import Header from '../../app/components/Header';

const props = {
  title: 'React & Redux TodoList',
  dispatch: fn => x => fn(x)
};

test('should render header', () => {
  const renderedComponent = renderer.create(
    <Header {...props} />
  );

  expect(renderedComponent.toJSON()).toMatchSnapshot();
});
