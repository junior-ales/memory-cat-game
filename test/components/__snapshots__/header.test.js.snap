exports[`test should render header 1`] = `
<header>
  <h1>
    A Todo List: 
    React & Redux TodoList
  </h1>
  <div>
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</header>
`;
